#!/usr/bin/env python
PACKAGE = "bumblebob_lateral_lq_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("q1", double_t, 0, "The Q matrix gain parameter for the lateral error value",  1, 0, 1)
gen.add("q2", double_t, 0, "The Q matrix gain parameter for the heading error value",  0.0, 0, 1)
gen.add("q3", double_t, 0, "The Q matrix gain parameter for the change of lateral error value",  0.0, 0, 1)
gen.add("q4", double_t, 0, "The Q matrix gain parameter for the change of heading error value",  0.0, 0, 1)

gen.add("R_matrix", double_t, 0, "The Cost parameter",  1, 0.1, 10)

gen.add("look_front", int_t, 0, "The estimation for the area in front of the car ",  5, 0, 50)
gen.add("look_back", int_t, 0, "The estimation for the area behind the car ",  5, 0, 20)


exit(gen.generate(PACKAGE, "bumblebob_lateral_lq_controller", "Lqr"))
